interface/ceed-preconditioning.c

```diff
@@ -1078,16 +1078,16 @@ int CeedOperatorAssemblyDataCreate(Ceed ceed, CeedOperator op, CeedOperatorAssem
       switch (eval_mode) {
         case CEED_EVAL_NONE:
         case CEED_EVAL_INTERP:
-          CeedCall(CeedRealloc(num_eval_mode_in + 1, &eval_mode_in));
-          eval_mode_in[num_eval_mode_in] = eval_mode;
-          num_eval_mode_in += 1;
+          CeedCall(CeedRealloc(num_active_in + 1, &eval_mode_in));
+          eval_mode_in[num_active_in] = eval_mode;
+          num_active_in += 1;
           break;
         case CEED_EVAL_GRAD:
-          CeedCall(CeedRealloc(num_eval_mode_in + dim, &eval_mode_in));
+          CeedCall(CeedRealloc(num_active_in + dim, &eval_mode_in));
           for (CeedInt d = 0; d < dim; d++) {
-            eval_mode_in[num_eval_mode_in + d] = eval_mode;
+            eval_mode_in[num_active_in + d] = eval_mode;
           }
-          num_eval_mode_in += dim;
+          num_active_in += dim;
           break;
```

Comment:
I think I'd still use 4 as the number of 'active inputs' here. (If you have an idea for a better phrase?) In the assembly, we handle each direction of the gradient separately

Reply:
Yeah, I understand that, but do we need to allocate 2 bases or 4 here? 
```c
CeedCall(CeedCalloc(2, &active_basis_in));
```
The idea was to copy the active basis using `CeedCall(CeedBasisReferenceCopy(basis_out, &(*data)->basis_out));`. I guess we need to copy 2 bases, right?

Reply:
No, that would bring us right back to the problem of messy indexing to figure out what basis were on. 

This seems to be getting a lot messier than I thought it would be.

Reply:
No, that would bring us right back to the problem of messy indexing to figure out what basis were on. 

This seems to be getting a lot messier than I thought it would be.

Reply:
I think I know what's going on, I'll update it next week.